opentracing-api/src/main/java/io/opentracing/Span.java

```diff
@@ -161,4 +161,21 @@
      * @see Span#context()
      */
     void finish(long finishMicros);
+
+    /**
+     * Returns an object that is an instance of the given class to allow access to non-standard methods.
+     *
+     * <p> Using this method is highly preferred to type casts, especially because there might be a hierarchy of wrapper
+     * objects. </p>
```

Comment:
based on the two implementations (Mock and Noop), not clear how this is different from cast - is the intention that wrappers will override this method?

Reply:
Yes, a wrapper's implementation might look like this:

```
public <T extends Span> T unwrap(Class<T> clazz) {
	if (clazz.isInstance(delegate)) {
		return clazz.cast(delegate);
	} else {
		return delegate.unwrap(clazz);
	}
}
```

Reply:
Or, if Java 8+ is the baseline, the interface can implement a basic `unwrap` routine, freeing up implementations to have code that would be mostly duplicated.